### PR TITLE
Added transactions/sync endpoint to get transaction data onto profile and cleaned up issues pointed out in past PRs

### PR DESCRIPTION
**Pull request by carsonwolber**
– Changed the schema for Users to include a column for access_token and 
changed transactions to decimal type instead of float to avoid floating point issues
- Moved back4app api request to designated helper
- Added transactions/sync endpoint and propagated the returned transactions json up to the user’s profile where the data can now be analyzed
- Fixed small issues and issues pointed out in other prs, spacing / comment issues, removing unused code, adding missing specification to helpers, and cleaning up print/log statements

See the transaction json as logged in the profile page (this has since been removed for the PR) here: 

Screenshot 2024-07-01 at 11 31 29 PM

### CODE REVIEW

**Review by pangers — APPROVED** [review]
Looks good to me!

File: frontend/HelperFuncs/plaidHelp.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ public_token })
+    });
+    if (!response.ok) {
+      throw new Error('Network response was not ok');
+    }
+    const data = await response.json();
+    console.log("access data", data);
```
Comment (carsonwolber) [inline frontend/HelperFuncs/plaidHelp.js:29]: I just saw I didn't save this file before pushing so this line's deletion won't show up until the next pr

File: frontend/HelperFuncs/utils.js [hunk 2]
```diff
@@ -15,18 +14,86 @@ export const submitProfile = async (userData) => {
...
 }
 
 
+/* 
+helper function to make put request to update user data with profile info 
+from account creation form
+*/
+export const populateAccount = async (userData, userId) => { 
+    try {
+        const response = await fetch(`${import.meta.env.VITE_BACKEND_ADDRESS}/profiles/${userId}`, {
```
Comment (lamchiminhmark) [inline frontend/HelperFuncs/utils.js:31]: unused variable. Make this
- `const _ = await ....`, or 
- `await ....`

Comment (carsonwolber) [reply frontend/HelperFuncs/utils.js:31]: good point, I will include that as a change in my next pr

**Review by William-La — APPROVED** [review]
Looks good--great work Carson!

**Pull request merged**